src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/finance/FinanceResourceIntTest.kt

```diff
@@ -20,74 +23,181 @@ class FinanceResourceIntTest : IntegrationTestBase() {
+        .jsonPath("$[0].postingType").isEqualTo(transaction1.postingType.name)
+        .jsonPath("$[0].description").isEqualTo(transaction1.entryDescription!!)
+        .jsonPath("$[0].entryDate").isEqualTo(transaction1.entryDate)
+        .jsonPath("$[0].clientReference").isEqualTo(transaction1.clientUniqueRef!!)
+        .jsonPath("$[0].reference").isEqualTo(transaction1.transactionReferenceNumber!!)
+        .jsonPath("$[0].subAccountType").isEqualTo(transaction1.subAccountType.name)
+        .jsonPath("$[0].generalLedgerTransactions[0].transactionId").isEqualTo(glTransaction1.transactionId)
+        .jsonPath("$[0].generalLedgerTransactions[0].transactionEntrySequence")
+        .isEqualTo(glTransaction1.transactionEntrySequence)
+        .jsonPath("$[0].generalLedgerTransactions[0].accountCode").isEqualTo(glTransaction1.accountCode.accountCode)
+        .jsonPath("$[0].generalLedgerTransactions[1].transactionId").isEqualTo(glTransaction2.transactionId)
+        .jsonPath("$[0].generalLedgerTransactions[1].transactionEntrySequence")
+        .isEqualTo(glTransaction2.transactionEntrySequence)
+        .jsonPath("$[0].generalLedgerTransactions[1].accountCode").isEqualTo(glTransaction2.accountCode.accountCode)
     }
 
-    with(
-      repository.getTransaction(OffenderTransaction.Companion.Pk(transaction.transactionId, 1))!!,
-    ) {
-      assertThat(transactionId).isEqualTo(transaction.transactionId)
-      assertThat(transactionEntrySequence).isEqualTo(1)
-      assertThat(offenderBooking?.bookingId).isEqualTo(transaction.offenderBooking!!.bookingId)
-      assertThat(trustAccount.prisonId).isEqualTo(transaction.trustAccount.prisonId)
-      assertThat(trustAccount.offenderId).isEqualTo(transaction.trustAccount.offenderId)
-      assertThat(trustAccount.accountClosed).isFalse()
-      assertThat(trustAccount.holdBalance).isCloseTo(transaction.trustAccount.holdBalance, Percentage.withPercentage(0.1))
-      assertThat(trustAccount.currentBalance).isCloseTo(transaction.trustAccount.currentBalance, Percentage.withPercentage(0.1))
-      assertThat(subAccountType).isEqualTo(transaction.subAccountType)
-      assertThat(transactionType).isEqualTo(transaction.transactionType)
-      assertThat(transactionReferenceNumber).isEqualTo(transaction.transactionReferenceNumber)
-      assertThat(clientUniqueRef).isEqualTo(transaction.clientUniqueRef)
-      assertThat(entryDate).isEqualTo(transaction.entryDate)
-      assertThat(entryDescription).isEqualTo(transaction.entryDescription)
-      assertThat(entryAmount).isCloseTo(transaction.entryAmount, Percentage.withPercentage(0.1))
-      assertThat(postingType).isEqualTo(transaction.postingType)
+    @Test
+    fun `none found`() {
+      webTestClient.get().uri("/transactions/9999")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_TRANSACTIONS")))
+        .exchange()
+        .expectStatus()
+        .isOk
+        .expectBody()
+        .json("[]")
```

Comment:
Does this need a not found test too (where transaction ID doesn't exist)?

Reply:
This is it, i.e. no records with this transaction id